include/guess/string_util.h

```diff
@@ -56,19 +53,26 @@ inline void normalize(std::string& s) {
   replace_all(s, "(", " ");
   replace_all(s, ")", " ");
 
-  for (int i = 0; i < s.length(); ++i) {
-    char c = s[i];
-    bool is_number = c >= '0' && c <= '9';
-    bool is_lower_case_char = c >= 'a' && c <= 'z';
-    bool is_upper_case_char = c >= 'A' && c <= 'Z';
-    if (!is_number && !is_lower_case_char && !is_upper_case_char) {
-      s[i] = ' ';
-    }
+  auto options = static_cast<utf8proc_option_t>(
+      utf8proc_option_t::UTF8PROC_DECOMPOSE |
+      utf8proc_option_t::UTF8PROC_STRIPMARK |
+      utf8proc_option_t::UTF8PROC_CASEFOLD);
+
+  utf8proc_uint8_t *utf8_out = nullptr;
+
+  // This is only safe when the representation of char matches unsigned char.
+  // Let's hope that's the case on all architectures we care about
+  auto *utf8_in = reinterpret_cast<const utf8proc_uint8_t *>(s.c_str());
+
+  size_t size = utf8proc_map(utf8_in, s.length(), &utf8_out, options);
+  if (size > 0) {
+    s = std::string(reinterpret_cast<const char *>(utf8_out), size);
+    free(utf8_out);
+  } else {
+    throw std::runtime_error{utf8proc_errmsg(size)};
```

Comment:
Would this prevent MOTIS from starting in case a single station name is not valid UTF-8? Maybe a more graceful option would be to log it and go back to the old procedure?

Reply:
The main repository redirecty `std::clog` to the module log file during import. So just using `std::clog` should help in this case.

https://github.com/motis-project/motis/blob/master/base/module/src/clog_redirect.cc

Reply:
But the `guesser` module doesn't create the lookup data structure during import. It's doing this in the `init` function. So it won't help much. I think it's still good with `std::clog` for now. At one point it will make sense to have a centralized logging via Open Telemetry or something like that. But that's not available yet.